contracts/Core/BlockManager.sol

```diff
@@ -120,24 +124,30 @@ contract BlockManager is Initializable, BlockStorage, StateManager, BlockManager
         require(stakerId > 0, "Structs.Staker does not exist");
         require(blocks[epoch].proposerId == 0, "Block already confirmed");
 
-        uint16[] memory deactivatedCollections = assetManager.getPendingDeactivations();
         if (sortedProposedBlockIds[epoch].length == 0 || blockIndexToBeConfirmed == -1) {
-            assetManager.executePendingDeactivations(epoch);
+            uint32 updateRegistryEpoch = collectionManager.getUpdateRegistryEpoch();
+            // slither-disable-next-line incorrect-equality
+            if (updateRegistryEpoch == epoch) {
+                collectionManager.updateRegistry();
+            }
             return;
         }
         uint32 proposerId = proposedBlocks[epoch][sortedProposedBlockIds[epoch][uint8(blockIndexToBeConfirmed)]].proposerId;
         require(proposerId == stakerId, "Block Proposer mismatches");
-        _confirmBlock(epoch, deactivatedCollections, proposerId);
+        _confirmBlock(epoch, proposerId);
     }
 
     function confirmPreviousEpochBlock(uint32 stakerId) external override initialized onlyRole(BLOCK_CONFIRMER_ROLE) {
         uint32 epoch = _getEpoch(epochLength);
-        uint16[] memory deactivatedCollections = assetManager.getPendingDeactivations();
         if (sortedProposedBlockIds[epoch - 1].length == 0 || blockIndexToBeConfirmed == -1) {
-            assetManager.executePendingDeactivations(epoch);
+            uint32 updateRegistryEpoch = collectionManager.getUpdateRegistryEpoch();
```

Comment:
registry needs to be updated irrespective of valid block. hence, shift it outside the if block